src/app/public/modules/modal/modal.service.ts

```diff
@@ -80,13 +97,17 @@ export class SkyModalService {
 
   private createHostComponent() {
     if (!SkyModalService.host) {
-      let factory = this.resolver.resolveComponentFactory(SkyModalHostComponent);
+      const componentRef = this.resolver
+        .resolveComponentFactory(SkyModalHostComponent)
+        .create(this.injector);
 
-      this.adapter.addHostEl();
+      const domElem = (componentRef.hostView as EmbeddedViewRef<any>).rootNodes[0];
```

Comment:
Just by way of recording our offline conversation, we're planning on implementing the new `SkyDynamicComponentService`, here.